### PR TITLE
perf: improve fps in technology picker screen by adding culling

### PR DESCRIPTION
**Pull request by shwwwa**
I did this pull request because my brother complained about low fps in technology screen on his low-end phone.

Current (stable) implementation of tech picker redraws every tech button every frame, including the ones scrolled off screen.
Although technically `ScrollPane` already computes a correct `cullingArea`, it sets it only on its direct widget `topTable`, so culling gets no-opped because `Group.drawChildren` culls only direct children against that rectangle.
This commit moves the culling rectangle one level down - `techTable.draw()` copies `cullingArea` of `topTable` into its own so small (centered) mod trees cull.

Few notes about PR
- Why I am reading the offset from the `Cell` and not `this.x/this.y`
In `Group`, `drawChildren` bakes the scroll offset into each child's x/y for the duration of the `draw()` call. 
Reading `this.x/y` makes the cull window drift with the scroll so content of the window disappears.
  
- Why `setSize` on `lines` / `orderIndicators`
They hold absolutely-positioned children and report a 0x0 bounding box, so culling would drop each group wholesale the instant it scrolled past the origin. 
Sizing them to span the table keeps them drawn (their children have no culling area of their own).
I also have set `Touchable.disabled` for them - they cover the whole tree, so without it they would intercept clicks meant for the buttons under them.

After this change fps improved from 15 to 78 (measured via SurfaceFlinger timestamps).
Small trees are centered and don't scroll so everything stays inside the window.
Large/modded trees cull more, so they benefit even more. When `ScrollPane` sets no culling area, the code falls back to `cullingArea = null` (draw everything, in other words previous behavior).

---------------------------------
Although I tested this in real game and a few edge cases, please test and review this thoroughly because I am a little bit bad at Kotlin lol
I didn't bother searching through commit/PRs/issues history to see if this was dealed before, so I would like to have some information if this would given a thought before
@yairm210 @SomeTroglodyte
- shwwwa

### CODE REVIEW

File: core/src/com/unciv/ui/screens/pickerscreens/TechPickerScreen.kt [hunk 1]
```diff
@@ -55,7 +57,23 @@ class TechPickerScreen(
      *  leaving us the juicy small tech tree right in the center.
      */
     private val techTable = object : Table(){
-        override fun draw(batch: Batch?, parentAlpha: Float) = super.draw(batch, parentAlpha)
+        private val childCullingArea = Rectangle()
+
+        override fun draw(batch: Batch?, parentAlpha: Float) {
+            val parentTable = parent as? Table
```
Comment (yairm210) [inline core/src/com/unciv/ui/screens/pickerscreens/TechPickerScreen.kt]: I don't like this, just specify topTable directly - it's A. definitely there and B. definitely a table so it definitely has a culling area and then there is no "None" case at all

File: core/src/com/unciv/ui/screens/pickerscreens/TechPickerScreen.kt [hunk 2]
```diff
@@ -55,7 +57,23 @@ class TechPickerScreen(
...
+            val parentCulling = parentTable?.cullingArea
+            if (parentCulling == null) {
+                cullingArea = null
+            } else {
+                val cell = parentTable.getCell(this)
+                val offsetX = cell?.actorX ?: 0f
+                val offsetY = cell?.actorY ?: 0f
+                childCullingArea.set(parentCulling.x - offsetX, parentCulling.y - offsetY,
+                    parentCulling.width, parentCulling.height)
```
Comment (yairm210) [inline core/src/com/unciv/ui/screens/pickerscreens/TechPickerScreen.kt]: We're taking the parent culling rect as is but moving it by... our offset? What's this about?

**Comment by yairm210** [comment]
In general great initiative! I want this in :D 
The only thing I'm not convinced of is the location of the culling area for the child

File: core/src/com/unciv/ui/screens/pickerscreens/TechPickerScreen.kt (hunk 1 above)
Comment (shwwwa) [reply core/src/com/unciv/ui/screens/pickerscreens/TechPickerScreen.kt]: good call

**Comment by shwwwa** [comment]
thank you for making me feel stupid considering everything got simplified into one line
tested on target still works properly

**Comment by SomeTroglodyte** [comment]
Highly interesting! And no, with libGdx quirks, even one-liners can be far from trivial, so no need to feel stupid.

File: core/src/com/unciv/ui/screens/pickerscreens/TechPickerScreen.kt [hunk 3]
```diff
@@ -43,8 +44,8 @@ class TechPickerScreen(
     private var selectedTech: Technology? = null
     private var civTech: TechManager = civInfo.tech
     private var tempTechsToResearch: ArrayList<String>
-    private var lines = NonTransformGroup()
-    private var orderIndicators = NonTransformGroup()
+    private var lines = NonTransformGroup().apply { touchable = Touchable.disabled }
+    private var orderIndicators = NonTransformGroup().apply { touchable = Touchable.disabled }
```
Comment (SomeTroglodyte) [inline core/src/com/unciv/ui/screens/pickerscreens/TechPickerScreen.kt:48]: Actually makes me wonder whether maybe `NonTransformGroup` would be better off defaulting to `Touchable.disabled`- but too lazy to review all 15 uses.

File: core/src/com/unciv/ui/screens/pickerscreens/TechPickerScreen.kt [hunk 4]
```diff
@@ -55,7 +56,10 @@ class TechPickerScreen(
      *  leaving us the juicy small tech tree right in the center.
      */
     private val techTable = object : Table(){
-        override fun draw(batch: Batch?, parentAlpha: Float) = super.draw(batch, parentAlpha)
+        override fun draw(batch: Batch?, parentAlpha: Float) {
```
Comment (SomeTroglodyte) [inline core/src/com/unciv/ui/screens/pickerscreens/TechPickerScreen.kt:59]: Java library / kotlin curiosity: `override fun draw(batch: Batch, parentAlpha: Float)` is perfectly safe, kotlin just does not trust the absence of the voluntary `@Null` annotation on the Java side as meaning the parameter is actually non-nullable. But in Gdx, `@Null` or its absence is quite reliable. Kotlin itself will accept both, the IDE will generate the "safer" template. Not relevant as long as you just pass it back to Java as this one or `NonTransformGroup` do, but can save some effort.

**Comment by SomeTroglodyte** [comment]
Only criticism: Worth a short comment _in code_ with the reason.

**Review by yairm210 — APPROVED** [review]
